viewer/src/components/SpecialAction.vue

```diff
@@ -50,31 +50,39 @@ export default class SpecialAction extends Vue {
   action: string[];
 
   @Prop()
-  shortcut?: string;
+  planet: Planet;
 
   @Prop()
-  planet: Planet;
+  player: Player | null;
 
   onClick() {
     if (!this._highlighted) {
       this.$emit("click");
       return;
     }
-    this.$store.dispatch("gaiaViewer/specialActionClick", this.action.join(","));
+    this.$store.dispatch("gaiaViewer/specialActionClick", this.button);
   }
 
-  get tooltip() {
-    return this.board ? undefined : withShortcut(translateResources(this.rewards), this.shortcut);
+  get income() {
+    return this.action.join(",");
   }
 
-  get rewards() {
-    return new Event('>' + this.action[0]).rewards;
+  get gameData(): Engine {
+    return this.$store.state.gaiaViewer.data;
+  }
+
+  get button(): ButtonData {
+    if (this.board) {
+      return {} as ButtonData;
```

Comment:
I would just return `null` here, and in the template test `button && button.warning` and `button && button.tooltip`